google/datalab/bigquery/commands/_bigquery.py

```diff
@@ -279,63 +280,64 @@ def _get_query_argument(args, cell, env):
-  else:
-    query = google.datalab.bigquery.Query(cell_body, values=env)
-
-  count = args['count']
-  method = args['method']
-  if method == 'random':
-    sampling = google.datalab.bigquery.Sampling.random(percent=args['percent'], count=count)
-  elif method == 'hashed':
-    sampling = google.datalab.bigquery.Sampling.hashed(field_name=args['field'],
-                                            percent=args['percent'],
-                                            count=count)
-  elif method == 'sorted':
-    ascending = args['order'] == 'ascending'
-    sampling = google.datalab.bigquery.Sampling.sorted(args['field'],
-                                            ascending=ascending,
-                                            count=count)
-  elif method == 'limit':
-    sampling = google.datalab.bigquery.Sampling.default(count=count)
-  else:
-    sampling = google.datalab.bigquery.Sampling.default(count=count)
-
-  if query:
-    results = query.sample(sampling=sampling, dialect=args['dialect'],
-                           billing_tier=args['billing'])
-  elif view:
-    results = view.sample(sampling=sampling)
-  else:
-    results = table.sample(sampling=sampling)
+  query = _get_query_argument(args, cell_body, google.datalab.utils.commands.notebook_environment())
   if args['verbose']:
-    print(results.sql)
-  if args['profile']:
-    return google.datalab.utils.commands.profile_df(results.to_dataframe())
-  else:
-    return results
+    print(query.sql)
+
+  if args['command'] in ['execute', 'extract']:
+    if args['command'] == 'execute':
+      if args['to_dataframe']:
+        # re-parse the int arguments because they're passed as strings
+        start_row = int(args['dataframe_start_row']) if args['dataframe_start_row'] else None
+        max_rows = int(args['dataframe_max_rows']) if args['dataframe_max_rows'] else None
+        output_options = QueryOutput.dataframe(start_row=start_row, max_rows=max_rows,
+                                               use_cache=not args['nocache'])
+      else:
+        output_options = QueryOutput.table(name=args['table'], mode=args['mode'],
+                                           use_cache=not args['nocache'],
+                                           allow_large_results=args['large'])
+    elif args['command'] == 'extract':
+      output_options = QueryOutput.file(path=args['path'], format=args['format'],
+                                        csv_delimiter=args['delimiter'],
+                                        csv_header=args['header'], compress=args['compress'],
+                                        use_cache=not args['nocache'])
+
+    r = query.execute(output_options, dialect=args['dialect'], billing_tier=args['billing'])
+    # if an execute was performed, a QueryJob is returned, which has a .results property
+    # otherwise, it's a base Job object, .result() should be called
+    return r.results if hasattr(r, 'results') else r.result()
```

Comment:
So based on your comments, if it returns a base Job object, even though it does not have .results field, hasattr(r, 'results') will be True because it has .results() method, and the function will return r.results which is a method?